### DISCUSSION TITLE
How is Open-Shell implemented/what entrypoint does Microsoft provide for creating a custom taskbar?

### DISCUSSION THREAD

**Discussion by expikr** [Open-Shell/Open-Shell-Menu#939] [Q&A]
Are there documented Windows API that Microsoft exposes for implementations like Open Shell?

**Comment by bynarie** [comment]
Yea I've been wanting to know the same thing. My best guess to alternative solutions, such as startallback for win11, is that it somehow "hides/disables" the original start menu and then puts a new button in the original location. So im guessing it does not directly hook to the start menu. I feel as if classicshell/openshell is a direct hook to the start menu.
